.github/workflows/docker-push.yml

```diff
@@ -0,0 +1,83 @@
+---
+name: 'Push Docker Image'
+on:
+  # TODO: remove, only for testing; replace with
+  # workflow_dispatch:
+  push:
+    branches:
+      - '**' # Trigger on every branch
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  dockerhub:
+    name: 'Build and Publish Docker Image'
+    runs-on: [self-hosted, linux, normal]
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.push.head.sha }}
```

Comment:
I would suggest we add `inputs` to `workflow_dispatch`:

```yml
on:
  workflow_dispatch:
    inputs:
      kontrol_branch:
        description: "Branch of kontrol to use to build the docker image"
        required: true
```

Then we change:

```
          ref: ${{ github.event.push.head.sha }}
```

to

```
          ref: ${{ github.events.inputs.kontrol_branch}}
```